lib/create-backoff.js

```diff
@@ -12,7 +12,10 @@ export default function createBackoff (maxRetries) {
   let attempt = 0
   return function maybeRetry (error, retry) {
     if (attempt < maxRetries) {
-      return promisedWait(Math.pow(2, attempt++) * 1000).then(retry)
+      let waitTime = (error.headers && error.headers['X-Contentful-RateLimit-Reset']) || Math.pow(2, attempt)
+      attempt++
+      waitTime *= 1000 * ((Math.random() * 1.2) + 1) // add a randomized buffer before recalling the api
```

Comment:
👍 for the comment but I think it should explain more why it's there.
I understand the code snippet quickly but the why is more important for me (our slack conversation).